dds_web/api/files.py

```diff
@@ -740,3 +740,22 @@ def put(self):
             db.session.commit()
 
         return {"message": "File info updated."}
+
+
+class UpdateFailedFiles(flask_restful.Resource):
+    """Add files from failed_delivery_log to DB using the "add_uploaded_files_to_db" utils function"""
+
+    @auth.login_required(role=["Unit Admin", "Unit Personnel", "Project Owner", "Researcher"])
+    @json_required
+    @handle_validation_errors
+    def put(self):
+        """Run flask command with failed_delivery_log."""
+
+        # Verify project ID and access
+        project = project_schemas.ProjectRequiredSchema().load(flask.request.args)
+
+        # Get the request json and pass it to add_uploaded_files_to_db
+        request_json = flask.request.get_json(silent=True)
+        dds_web.utils.add_uploaded_files_to_db(project, request_json)
+
+        return {"message": "File(s) added to database."}
```

Comment:
This never returns an error since the `add_uploaded_files_to_db` doesn't raise anything. Could for example return the errors from the function and then let the `update_uploaded_file_with_log` command log the errors like the function does now, and this endpoint would handle the errors in another way if there are any. Plus this way you shouldn't need to do the additional check in the CLI to see whether or not the files have been added or not.

Reply:
Do you think it is also good to return 'files_added' (returned from 'add_uploaded_files_to_db'), so that one can check this in CLI, or will it be enough if 'errors' is empty?
Right now, we will have 'errors' if file is not found in S3, or exists in DB, but if 'db.session.commit()' in the end fails we will have the file into 'files_added' and no 'errors[file]'

Reply:
something like this!

Reply:
Yes, this looks good!

And yes I think that sounds like a good idea! I think it should be enough to just return an empty dict in that case.
So have something like 

```python
        files_added, errors = dds_web.utils.add_uploaded_files_to_db(project, request_json)
        return {"files_added": files_added, "errors": errors}
```

Reply:
Sure, but it became
`return {"files_added": [file.name for file in files_added], "message": errors}`
because having "errors" in the returned value causes an error in client's data_putter